posts/Rust-1.86.0.md

```diff
@@ -0,0 +1,175 @@
+
+Inside of functions not marked with the target feature attribute they can be called inside of an `unsafe` block, however it is the callers responsibility to ensure that the target feature is available.
+
+```rust
+#[target_feature(enable = "avx2")]
+fn requires_avx2() {
+    // ... snip
+}
+
+#[target_feature(enable = "avx2")]
+fn safe_callsite() {
+    // Calling `requires_avx2` here is safe as `bar`
+    // requires the `avx2` feature itself.
+    requires_avx2();
+}
+
+fn unsafe_callsite() {
+    // Calling `requires_avx2` here is unsafe, as we must
+    // ensure that the `avx2` feature is available first.
+    if is_x86_feature_detected!("avx2") {
+        unsafe { requires_avx2() };
+    }
+}
+```
+
+You can check the [`target_features_11`](https://github.com/rust-lang/rfcs/blob/master/text/2396-target-feature-1.1.md) RFC for more information.
+
+### Debug assertions that pointers are non-null when required for soundness
+
+The compiler will now insert debug assertions that a pointer is not null upon non-zero-sized reads and writes, and also when the pointer is reborrowed into a reference. For example, the following code will now produce a non-unwinding panic when debug assertions are enabled:
+```rust
+let _x = *std::ptr::null::<u8>();
+let _x = &*std::ptr::null::<u8>();
+```
+Trivial examples like this have produced a warning since Rust 1.53.0, the new runtime check will detect these scenarios regardless of complexity.
+
+These assertions only take place when debug assertions are enabled which means that they **must not** be relied upon for soundness. This also means that dependencies which have been compiled with debug assertions disabled (e.g. the standard library) will not trigger the assertions even when called by code with debug assertions enabled.
+
+### Make `missing_abi` lint warn by default
+
+Omitting the ABI in extern blocks and functions (e.g. `extern {}` and `extern fn`) will now result in a warning (via the `missing_abi` lint). Omitting the ABI after the `extern` keyword has always implicitly resulted in the `"C"` ABI. It is now recommended to explicitly specify the `"C"` ABI (e.g. `extern "C" {}` and `extern "C" fn`).
+
+You can check the [Explicit Extern ABIs RFC](https://rust-lang.github.io/rfcs/3722-explicit-extern-abis.html) for more information.
+
+### Target deprecation warning for 1.87.0
+
+The target `i586-pc-windows-msvc` will be removed in the next version of Rust, 1.87.0. Its difference to the much more popular `i686-pc-windows-msvc` is that it does not require SSE2 instruction support, but Windows 10, the minimum required OS version of all `windows` targets (except the `win7` targets), requires SSE2 instructions itself.
+
+All users currently targeting `i586-pc-windows-msvc` should migrate to `i686-pc-windows-msvc` before the `1.87.0` release.
+
+You can check the [Major Change Proposal](https://github.com/rust-lang/compiler-team/issues/840) for more information.
+
+### Stabilized APIs
```

Comment:
FYI, the original order of items seen here was whatever the relnotes tool generated, but I tweaked that for the final relnotes PR to make more sense (IMO). So you may want to repeat that change here.